File: scala-core/src/main/scala/idesyde/core/headers/DesignModelHeader.scala
package idesyde.core.headers

import upickle.default.*

case class DesignModelHeader(
    val category: String,
    val model_paths: Set[String],
    val elements: Set[String],
    val relations: Set[LabelledArcWithPorts]
) derives ReadWriter {

    def asText: String = write(this)

    def asBinary: Array[Byte] = writeBinary(this)
}

File: kotlin-core/src/main/kotlin/idesyde/core/headers/DesignModelHeader.kt
package idesyde.core.headers

import kotlinx.serialization.*
@Serializable
data class DesignModelHeader (
        val category: String,
        val elements: List<String>,

        @SerialName("model_paths")
        val modelPaths: List<String>,

        val relations: List<LabelledArcWithPorts>
)
